llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp

```diff
@@ -1166,7 +1166,7 @@ static Value *extractIntPart(const IntPart &P, IRBuilderBase &Builder) {
     V = Builder.CreateLShr(V, P.StartBit);
   Type *TruncTy = V->getType()->getWithNewBitWidth(P.NumBits);
   if (TruncTy != V->getType())
-    V = Builder.CreateTrunc(V, TruncTy);
+    V = Builder.CreateZExtOrTrunc(V, TruncTy);
```

Comment:
It is confusing to me. Can you explain this further?

Reply:
Was incorrectly setting a field. Fixed :)